kernel/CMakeLists.txt

```diff
@@ -117,6 +117,35 @@ target_sources_ifdef(CONFIG_OBJ_CORE              kernel PRIVATE obj_core.c)
 
 if(${CONFIG_KERNEL_MEM_POOL})
   target_sources(kernel PRIVATE mempool.c)
+
+  # Import all custom HEAP_MEM_POOL minimum size requirements
+  import_kconfig(CONFIG_HEAP_MEM_POOL_MIN_SIZE_ ${DOTCONFIG} min_size_keys)
+
+  set(max_min_size 0)
+
+  foreach(min_size ${min_size_keys})
+    if(${${min_size}} GREATER ${max_min_size})
+      set(max_min_size ${${min_size}})
+    endif()
+  endforeach()
+
+  if(CONFIG_HEAP_MEM_POOL_SIZE LESS max_min_size)
+    # Only warn if default value 0 has been modified
+    if(NOT CONFIG_HEAP_MEM_POOL_SIZE EQUAL 0)
+      message(WARNING "
+      CONFIG_HEAP_MEM_POOL_SIZE is less than HEAP_MEM_POOL_MIN_SIZE:
+        ${CONFIG_HEAP_MEM_POOL_SIZE} < ${max_min_size}")
+    endif()
+    if(CONFIG_HEAP_MEM_POOL_IGNORE_MIN)
+      set(max_min_size ${CONFIG_HEAP_MEM_POOL_SIZE})
+    endif()
+  else()
+    set(max_min_size ${CONFIG_HEAP_MEM_POOL_SIZE})
+  endif()
+
+  zephyr_compile_definitions(K_HEAP_MEM_POOL_SIZE=${max_min_size})
+else()
+  zephyr_compile_definitions(K_HEAP_MEM_POOL_SIZE=0)
```

Comment:
This else is taken when `CONFIG_HEAP_MEM_POOL_SIZE==0`.

Basically this requires that a lib / subsys is which does `HEAP_MEM_POOL_SIZE_FOO=<value>` is also required to ensure `CONFIG_HEAP_MEM_POOL_SIZE!=0`.

Is that intentionally ?

Reply:
This is the else-branch for `if(${CONFIG_KERNEL_MEM_POOL})`, in which case the entire `kernel/mempool.c` will not be included in the build.

Reply:
> This is the else-branch for `if(${CONFIG_KERNEL_MEM_POOL})`, in which case the entire `kernel/mempool.c` will not be included in the build.

That said, maybe we want a warning if any non-zero `HEAP_MEM_POOL_MIN_SIZE_FOO` options exist but the user has disabled `KERNEL_MEM_POOL` (it defaults to enabled).

Reply:
> > This is the else-branch for `if(${CONFIG_KERNEL_MEM_POOL})`, in which case the entire `kernel/mempool.c` will not be included in the build.
> 
> That said, maybe we want a warning if any non-zero `HEAP_MEM_POOL_MIN_SIZE_FOO` options exist but the user has disabled `KERNEL_MEM_POOL` (it defaults to enabled).

Actually, not quite sure about this. The current idea is that `kernel/CMakeLists.txt` doesn't even both looking for and parsing MIN_SIZE options if `KERNEL_MEM_POOL` is not set (the idea was to avoid the build time overhead if it serves no useful purpose)

Reply:
And I just realised that `CONFIG_HEAP_MEM_POOL_SIZE` till be unset if `CONFIG_KERNEL_MEM_POOL` is unset (since the latter depends on the former), so we should probably follow the same for `K_HEAP_MEM_POOL_SIZE`, i.e. not define it at all in that case.